database/pebble/iterator.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package pebble
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/cockroachdb/pebble"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+var (
+	_ database.Iterator = (*iter)(nil)
+
+	errCouldntGetValue = errors.New("couldnt get iterator value")
+	errNoPointKey      = errors.New("iterator has no point key")
+)
+
+type iter struct {
+	// [lock] ensures that only one goroutine can access [iter] at a time.
+	// Note that [Database.Close] calls [iter.Release] so we need [lock] to ensure
+	// that the user and [Database.Close] don't execute [iter.Release] concurrently.
+	// Invariant: [Database.lock] is never grabbed while holding [lock].
+	lock sync.Mutex
```

Comment:
nit: This could be a `RWMutex`... I think it's fine as is because iterators are documented as not being thread safe though.